### DISCUSSION TITLE
OpenSUSE Leap Install Instructions.

### DISCUSSION THREAD

**Discussion by mnavarro83** [OctoFarm/OctoFarm#745] [General]
I had no issues getting this running on OpenSUSE and I figured I'd share the steps as a small part of my giving something back to the community. If any other distros want instructions I will be happy to write them up:

1) Update your repositories and packages fully:
`    sudo zypper dup`

2) Install necessary dependencies:
`    sudo zypper install nodejs gcc make git`

3) Verify correct Nodejs version (needs to be 14+)
`    node --version`

4) Import the necessary pgp key and add the MongoDB community repository
```
    sudo rpm --import https://www.mongodb.org/static/pgp/server-5.0.asc
    sudo zypper addrepo --gpgcheck "https://repo.mongodb.org/zypper/suse/15/mongodb-org/5.0/x86_64/" mongodb
```
 
5) Install, start and make MongoDB persist reboots:
```
    sudo zypper install mongodb-org
    sudo systemctl enable mongod --now
    sudo systemctl status mongod
```
    
6) Install pm2:
`    sudo npm install pm2 -g`

7) Obtain and start OctoFarm
```
    git clone --depth 1 https://github.com/OctoFarm/OctoFarm.git
    cd OctoFarm/
    npm start
    pm2 list
```

8) Allow traffic on port 4000 during the current boot:
```
    sudo firewall-cmd --add-port=4000/tcp
    sudo firewall-cmd --add-port=4000/udp
```

9) And finally with one additional flag to make it persist reboots
```
    sudo firewall-cmd --add-port=4000/udp --permanent
    sudo firewall-cmd --add-port=4000/tcp --permanent
```

That's it. Navigate to <printer_ip>:4000 in your browser and have a day!

**Comment by NotExpectedYet** [comment]
Legend thanks for these! Would it be ok to add to the docs repo? Your welcome to do a PR for it, or I can add it no trouble.

**Reply by mnavarro83** [reply]
Absolutely. I'll look into doing a PR for that now. I'll admit that I am a bit of a Github noob when it comes to contributing (I've used it purely selfishly so far) so bear with me... If instructions for any other Linux distros are wanted I am more than happy to go through the installation process, verify functionality/compatability and get them written.

**Reply by NotExpectedYet** [reply]
No worries! Big thanks. I'm happy for any additions there as I only use Debian/Ubuntu distributions personally so don't have the experience with others.

**Reply by mnavarro83** [reply]
Awesome, PR done. I'll work on instructions for Arch and Fedora/Alma/Rocky this week then and that should give us a pretty broad spectrum. We will have covered Cannonical, Redhat, OpenSUSE and Arch... anyone running something different is either smart enough to piece the install together based on that documentation or is unhelp-able lol.

**Reply by NotExpectedYet** [reply]
Epic I'll take a look sometime today and get it commited. Thanks for the effort!
